otcextensions/osclient/dis/v2/data.py

```diff
@@ -0,0 +1,248 @@
+                   "request."
+                   "\nNote: If the value is less than the size of a single "
+                   "record in the partition, the record cannot be obtained."),
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.dis
+        attrs = {
+            'partition-cursor': parsed_args.partition_cursor
+        }
+        if parsed_args.max_fetch_bytes:
+            attrs.update(max_fetch_bytes=parsed_args.max_fetch_bytes)
+
+        data = client.download_data(**attrs)
+
+        return (
+            self.display_columns,
+            (
+                utils.get_item_properties(
+                    s, self.columns, formatters=_formatters
+                ) for s in data
+            )
+        )
+
+
+class ShowDataCursor(command.ShowOne):
+    _description = _("Query Data Cursor.")
+
+    def get_parser(self, prog_name):
+        parser = super(ShowDataCursor, self).get_parser(prog_name)
+        parser.add_argument(
+            'stream',
+            metavar='<stream>',
+            help=_("Name of the stream."),
+        )
+        parser.add_argument(
+            '--partition-id',
+            metavar='<partition_id>',
+            dest='partition-id',
+            required=True,
+            help=_("Partition ID of the stream."),
+        )
+        parser.add_argument(
+            '--cursor-type',
+            dest='cursor-type',
+            metavar='{' + ','.join(CURSOR_TYPE_CHOICES) + '}',
+            type=lambda s: s.upper(),
+            choices=CURSOR_TYPE_CHOICES,
+            help=_("Cursor Type."),
+        )
+        parser.add_argument(
+            '--starting-seq-num',
+            metavar='<starting_seq_num>',
+            dest='starting-sequence-number',
+            help=_("Sequence number. A sequence number is the unique "
+                   "identifier of each record."),
+        )
+        parser.add_argument(
+            '--timestamp',
+            metavar='<timestamp>',
+            type=int,
+            help=_("Timestamp when the data record starts to be read, which "
+                   "is closely related to cursor type `AT_TIMESTAMP`."),
+        )
+        parser.add_argument(
+            '--stream-id',
+            metavar='<stream_id>',
+            dest='stream-id',
+            help=_("Unique ID of the stream."),
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        client = self.app.client_manager.dis
+        attrs = {
+            'stream-name': parsed_args.stream
+        }
+        for arg in ('partition-id', 'cursor-type',
+                    'starting-sequence-number',
+                    'timestamp', 'stream-id',):
+            val = getattr(parsed_args, arg)
+            if val:
+                attrs[arg] = val
+
+        obj = client.get_data_cursor(**attrs)
+        display_columns, columns = _get_columns(obj)
```

Comment:
if i understand correctly _get_columns  returns display names which are not is such beautiful form you are using in another cases, do we have different output for different commands then?